src/glutil.h

```diff
@@ -105,6 +107,46 @@ inline void glLoadMatrix(const Imath::M44f& m)
     glLoadMatrixf((GLfloat*)m[0]);
 }
 
+//------------------------------------------------------------------------------
+// Texture utility
+
+struct Texture
+{
+    Texture(const QImage& i)
+    :   image(i),
+        target(GL_TEXTURE_2D),
+        texture(0)
+    {
```

Comment:
Good catch!  Yes, it makes sense to check the format.

Seems kinda lame to need a QGLContext reference to bind a texture, I'd put that down to an OpenGL/Qt paradigm mismatch.  I'd rather skip that path, unless you feel strongly about it.

I pushed a fixup for checking the format.  On the conversion, I generally wouldn't expect a 1-channel texture to be bloated to RGBA by default, so I'd be a bit more inclined to have an assertion there.